### PR TITLE
1.21.0 release notes

### PR DESCRIPTION
**Pull request by keirthana**
This is the <s>first</s> draft of 1.21.0 release notes. Pending changes and doubts are marked in comments.

Pending tasks:

- [ ] Add links to new topics
    - [ ] VHAL and AAOS
    - [x] Graphics debugging with renderdoc
    - [x] How to enable VirGL

- [x] Add NATS charm link to upgrade instructions once @jat-canonical has updated it.

### CODE REVIEW

**Comment by keirthana** [comment]
The following Jira tickets are not yet added as I need more clarification on them, I will discuss with the team and then add them:
* [AC-2220](https://warthogs.atlassian.net/browse/AC-2220)
* [AC-2167](https://warthogs.atlassian.net/browse/AC-2167)
* [AC-2093](https://warthogs.atlassian.net/browse/AC-2093)

File: reference/release-notes/1.21.0.md
Comment (keirthana) [inline reference/release-notes/1.21.0.md:54]: @morphis Please add the known issue.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [inline reference/release-notes/1.21.0.md:41]: I have updated the current WebView picked by 1.21/stable. I will update this if and when it changes.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [inline reference/release-notes/1.21.0.md:40]: I will add the link after the security bulletin is published.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [inline reference/release-notes/1.21.0.md:39]: @ajanon Once documentation for VHAL is published, I will add those links as cross-references here. But this may have to happen on the release day as I assume there will be new topics for VHAL documentation.

File: reference/release-notes/1.21.0.md
Comment (adglkh) [inline reference/release-notes/1.21.0.md:17]: I feel like we need a dedicated page to introduce this new feature like what we did in our internal doc
https://github.com/canonical/anbox-internal-docs/blob/main/howto/anbox/android-tracing.md
Because currently, Android system tracing support is gated by the `android.enable_tracing` feature flag. Route the end user to  [System tracing overview](https://developer.android.com/topic/performance/tracing) without enabling this feature may not help

File: reference/release-notes/1.21.0.md
Comment (adglkh) [inline reference/release-notes/1.21.0.md:63]: A few are missing
- https://warthogs.atlassian.net/browse/AC-2255
- https://warthogs.atlassian.net/browse/AC-2256

File: reference/release-notes/1.21.0.md
Comment (adglkh) [inline reference/release-notes/1.21.0.md:61]: Let's drop this as the fix has been landed in the 1.20.2 release.
https://discourse.ubuntu.com/t/anbox-cloud-1-20-2-has-been-released/41686

File: reference/release-notes/1.21.0.md
Comment (adglkh) [reply reference/release-notes/1.21.0.md:41]: The webview version has been bumped up to `121.0.6167.101` a few days ago for stable-1.21 branch

File: reference/release-notes/1.21.0.md
Comment (adglkh) [reply reference/release-notes/1.21.0.md:54]: Let's add the following issue 
https://warthogs.atlassian.net/browse/AC-1995

For the virgl feature, from my testing with the latest Android 12 iamge, the graphics stuttering and flicking issues happen from time to time as well.  I'll let Simon comment this if we want to put this in this section.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md:63]: The first two bullet points are AC-2256 and AC-2255 respectively. Please let me know if we need to add more details.

File: reference/release-notes/1.21.0.md
Comment (adglkh) [reply reference/release-notes/1.21.0.md:17]: I think we can but it might require some wording adjustments.
@morphis What do you think?

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Done

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md:17]: Wording adjustments are fine, I was asking about technical accuracy. I will look to use the internal docs as a base and create a separate PR for that.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Thanks for the catch!

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md:17]: I created a PR for this: https://github.com/canonical/anbox-cloud-docs/pull/34

Comment (adglkh) [reply reference/release-notes/1.21.0.md]: @keirthana, @morphis  and I had a quick discussion about it. Let's include that in the release announcement but hold off on creating a dedicated page to introduce this feature, as it may require some polishing to improve the overall user experience for Android system tracing

File: reference/release-notes/1.21.0.md
Comment (adglkh) [reply reference/release-notes/1.21.0.md:64]: Ah, Okay, thanks for pointing this out.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: okay, I will then leave the PR on that open for now and will just file it for later once I am back.

File: reference/release-notes/1.21.0.md
Comment (jat-canonical) [inline reference/release-notes/1.21.0.md:22]: [AC-1700](https://warthogs.atlassian.net/browse/AC-1700) Was missed I think because of the missing label.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: @morphis So I take it that you'd rather not have this called out for 1.21?

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md:16]: This is done using https://github.com/canonical/anbox-cloud-docs/pull/35

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Confirmed not to add this for 1.21. See Jira comments for details.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Confirmed not to add this for now based on Simon's comments on https://github.com/canonical/anbox-cloud-docs/pull/34

Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Removed this.

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Added them now

File: reference/release-notes/1.21.0.md
Comment (morphis) [inline reference/release-notes/1.21.0.md:33]: We should clarify what these are or drop the point as it reads quite generic.

File: reference/release-notes/1.21.0.md
Comment (morphis) [inline reference/release-notes/1.21.0.md:29]: We should give an example what this means

File: reference/release-notes/1.21.0.md
Comment (morphis) [inline reference/release-notes/1.21.0.md:31]: We should give an example here on how the parameter can be used.

File: reference/release-notes/1.21.0.md
Comment (morphis) [inline reference/release-notes/1.21.0.md:32]: We should link VHAL to https://source.android.com/docs/automotive/vhal and also call out this is specific to AAOS

File: reference/release-notes/1.21.0.md
Comment (keirthana) [reply reference/release-notes/1.21.0.md]: I have explained the user impact more in detail.

**Comment by keirthana** [comment]
@morphis Updated the RN based on all that we discussed today. Pending changes are listed in description as a task list.

File: reference/release-notes/1.21.0.md
Comment (morphis) [inline reference/release-notes/1.21.0.md:50]: This should have a public bug.

Comment (keirthana) [reply reference/release-notes/1.21.0.md]: Created one - [LP 2053106](https://bugs.launchpad.net/anbox-cloud/+bug/2053106)

**Review by jat-canonical — APPROVED** [review]
LGTM

**Review by morphis — APPROVED** [review]
LGTM

**Pull request merged**
